zerver/views/muting.py

```diff
@@ -100,6 +106,8 @@ def mute_user(request: HttpRequest, user_profile: UserProfile, muted_user_id: in
         return json_error(_("User already muted"))
 
     do_mute_user(user_profile, muted_user, date_muted)
+    do_mark_user_messages_as_read(user_profile, muted_user)
```

Comment:
Let's make `do_mute_user` call this internally?  That's safer, since it handles the case where some other code path starts being able to mute users as well.